src/categorical_algebra/DataMigrations.jl

```diff
@@ -335,9 +340,36 @@ function (M::SigmaMigrationFunctor)(d::ACSet; n=100)
       res[domval,h] = codomval
     end
   end
+  #Go back and make sure attributes that ought to have 
+  #specific values because of d do have those values.
+  for (k,kdom,kcod) in attrs(S)
+    f = hom_map(F,k)
+    #split f into its hom part and its attr part
+    f1,f2 = split_r(f)
+    #Need f1 on the collage for rel_res but f2 
+    #on the target schema
+    f1 = hom_map(i2,f1)
+    for i in parts(d,kdom)
+      oldval = subpart(d,i,k)
+      src_a,tgt_a=add_srctgt(Symbol("α_$kdom"))
+      src_f,tgt_f=add_srctgt(Symbol("$f1"))
+      #Find where i goes under alpha, and then where that goes
+      #under the hom part of f, by walking the spans in rel_res.
+      j = rel_res[only(incident(rel_res,i,src_a)),
+                  tgt_a] 
+      f1j = f1 isa GATExpr{:id} ? j :
+       rel_res[only(incident(rel_res,j,src_f)),
+              tgt_f]
+      res[f1j,nameof(f2)] = oldval
+    end
+  end
   return res
 end 
-
+"""
+Split an n-fold composite (n may be 1) 
+Hom or Attr into its left n-1 and rightmost 1 components
+"""
+split_r(f) = f isa GATExpr{:compose} ? (compose(f.args[1:end-1]),f.args[end]) : (id(dom(f)),f)
```

Comment:
Better to use the accessor functions for GAT exprs, so this would be:

```julia
head(f) == :compose ? (compose(args(f)[1:end-1]), last(f)) : (id(dom(f)), f)
```